<?xml version="1.0" ?>
<launch>

  <!-- Set Robot model/rviz configuration path-->
  <arg name="urdf-name"                   default="kuka_lwr_lasa.urdf.xacro"/>
  <arg name="load-rviz"                   default="true"/>
  <arg name="rviz-config-path"            default="$(find kuka_lwr_bringup)/rviz-config/lwr-rviz-config.rviz"/>

  <!-- Load KUKA Realtime visualization -->
  <include file="$(find kuka_lwr_bringup)/launch/lwr_realtime.launch">
  <!-- Override Argument to desired robot model -->
    <arg name="urdf-name"                   value="$(arg urdf-name)"/>
    <arg name="load-rviz"                   value="$(arg load-rviz)"/>
    <arg name="rviz-config-path"            value="$(arg rviz-config-path)"/>    
  </include>
<!--
 BARRETT HAND  
  <group ns="bhand">
    <param name="robot_description"
    command="$(find xacro)/xacro.py '$(find barrett_hand_description)/robots/bh_alone.urdf.xacro'" />

  <include file="$(find barrett_hand_description)/launch/bh_description_arm.launch" />
  </group>
 -->  

  <!-- PR2 Loopback Controllers Controllers -->
  <include file="$(find kuka_lwr_bringup)/launch/controllers/lwr_loopback_sim_controllers.launch"/>
  
  <!-- IAI Velocity Controllers -->
  <include file="$(find kuka_lwr_bringup)/launch/controllers/lwr_sim_vel_controllers.launch"/>
  
  <!-- IAI Position Controllers -->
  <include file="$(find kuka_lwr_bringup)/launch/controllers/lwr_sim_pos_controllers.launch"/>

  <!-- Load LASA to IAI joint state parser -->
  <node name="lasa_to_iai_parser" pkg="lasa_to_iai_parser" type="lasa_to_iai.py" output="screen"/>

  <!-- Load the planner -->
  <node name="organizing_obj_task_motion_planner" pkg="organizing_obj_task_motion_planner" type="organizing_obj_task_motion_planner_py.py" output="screen"/>

  <!-- Load barrett hand controller  -->
  <node pkg="bhand_controller" type="bhand_node.py" name="bhand_node" output="screen">
      <rosparam command="load" file="$(find bhand_controller)/config/bhand.yaml" />
      <param name="topic_state" value="state" />
      <param name="port" value="/dev/pcanusb0" />
      <param name="tactile_sensors" value="True" />
  </node>
</launch>
